From 01d5c41eb3071d66c3cde1ec19d26a4c817c3464 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:27 -0400
Subject: [ethernet] i40e: Cleaning up missing null-terminate in conjunction with strncpy

Message-id: <1409658042-22689-267-git-send-email-sassmann@redhat.com>
Patchwork-id: 90450
O-Subject: [RHEL7.1 PATCH 266/281] i40e: Cleaning up missing null-terminate in conjunction with strncpy
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>

Replacing strncpy with strlcpy to avoid strings that lacks null terminate.

Signed-off-by: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
Tested-By: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 35a7d8045931d384c3d02f28287e0eb42d6038f4)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 8f33549..50751df 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4415,13 +4415,13 @@ static void i40e_print_link_message(struct i40e_vsi *vsi, bool isup)
 
  switch (vsi->back->hw.phy.link_info.link_speed) {
  case I40E_LINK_SPEED_40GB:
-  strncpy(speed, "40 Gbps", SPEED_SIZE);
+  strlcpy(speed, "40 Gbps", SPEED_SIZE);
   break;
  case I40E_LINK_SPEED_10GB:
-  strncpy(speed, "10 Gbps", SPEED_SIZE);
+  strlcpy(speed, "10 Gbps", SPEED_SIZE);
   break;
  case I40E_LINK_SPEED_1GB:
-  strncpy(speed, "1000 Mbps", SPEED_SIZE);
+  strlcpy(speed, "1000 Mbps", SPEED_SIZE);
   break;
  default:
   break;
@@ -4429,16 +4429,16 @@ static void i40e_print_link_message(struct i40e_vsi *vsi, bool isup)
 
  switch (vsi->back->hw.fc.current_mode) {
  case I40E_FC_FULL:
-  strncpy(fc, "RX/TX", FC_SIZE);
+  strlcpy(fc, "RX/TX", FC_SIZE);
   break;
  case I40E_FC_TX_PAUSE:
-  strncpy(fc, "TX", FC_SIZE);
+  strlcpy(fc, "TX", FC_SIZE);
   break;
  case I40E_FC_RX_PAUSE:
-  strncpy(fc, "RX", FC_SIZE);
+  strlcpy(fc, "RX", FC_SIZE);
   break;
  default:
-  strncpy(fc, "None", FC_SIZE);
+  strlcpy(fc, "None", FC_SIZE);
   break;
  }
 
@@ -5839,7 +5839,7 @@ static void i40e_send_version(struct i40e_pf *pf)
  dv.minor_version = DRV_VERSION_MINOR;
  dv.build_version = DRV_VERSION_BUILD;
  dv.subbuild_version = 0;
- strncpy(dv.driver_string, DRV_VERSION, sizeof(dv.driver_string));
+ strlcpy(dv.driver_string, DRV_VERSION, sizeof(dv.driver_string));
  i40e_aq_send_driver_version(&pf->hw, &dv, NULL);
 }
 
-- 
1.7.1